src/main/java/com/blibli/future/controller/ApprovalController.java

```diff
@@ -0,0 +1,60 @@
+package com.blibli.future.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.blibli.future.exception.IdNotFoundException;
+import com.blibli.future.service.api.ApprovalService;
+import com.blibli.future.vo.ApprovalRequestVo;
+import com.blibli.future.vo.ApprovalResponseVo;
+
+
+@RestController
+public class ApprovalController {
+	
+	public final String BASE_PATH = RequestController.BASE_PATH;
+	public final String PATH_APPR_OR_REJECT = BASE_PATH + "/{type}/{id}";
+	
+	private ApprovalService approvalService;
+	
+	@Autowired
+	ApprovalController(ApprovalService approvalService){
+		this.approvalService = approvalService;
+	}
+	
+	@PutMapping(PATH_APPR_OR_REJECT)
+	public ResponseEntity<String> approving(@PathVariable String type , @PathVariable String id , 
+			@RequestBody ApprovalRequestVo approvalRequestVo) throws IdNotFoundException{
+		if(type.equals("leave")){
+			approvalService.processLeave(id, approvalRequestVo.getNik(), approvalRequestVo.isApproved());
+		}
+		else if(type.equals("absence")){
+			approvalService.processAbsencePermit(id, approvalRequestVo.getNik(), approvalRequestVo.isApproved());
+		}
+		else{
+			throw new IdNotFoundException("Type: " + type + " was not found");
+		}
+		return new ResponseEntity<String>("id: " + id + " processed", HttpStatus.OK);
+	}
+	
+	@GetMapping(BASE_PATH)
+	public ResponseEntity<List<ApprovalResponseVo>> unapprovedList(@RequestParam String chiefNik, @RequestParam String type){
+		System.out.println(chiefNik);
+		if(type.equals("history")){
+			return new ResponseEntity<List<ApprovalResponseVo>>(approvalService.getRequestHistories(chiefNik), HttpStatus.OK);
+		}
+		else if(type.equals("unapproved")){
+			return new ResponseEntity<List<ApprovalResponseVo>>(approvalService.getUnapprovedRequests(chiefNik), HttpStatus.OK);
+		}
+		return new ResponseEntity(null, HttpStatus.BAD_REQUEST);
+	}
```

Comment:
Several question for this class:
1. what does `unapprovedList` means? Since, it's a method name, can we make it to more "verb-like", like `getUnapprovedRequests`?
2. (Suggestion) How about making exception for `unapprovedList`? something like `typeNotFoundException` and add it into the controller advice?